From e850373d3bd409e2ab04572deec3dbc546a0fc3d Mon Sep 17 00:00:00 2001
From: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Thu, 27 Apr 2023 11:07:51 -0300
Subject: [PATCH] csi-powerstore: add default values for templates

This patch consists of adding the default values in the values.yaml of
the helm chart.
It's necessary because the images in templates were in unexpected format
for the system and because templates need default values to be able
to build the package.

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Signed-off-by: Hediberto Cavalcante da Silva <hediberto.cavalcantedasilva@windriver.com>
---
 helm/csi-powerstore/values.yaml | 25 +++++++++++++++++++++----
 1 file changed, 21 insertions(+), 4 deletions(-)

diff --git a/helm/csi-powerstore/values.yaml b/helm/csi-powerstore/values.yaml
index c5aee2b..05e1c71 100644
--- a/helm/csi-powerstore/values.yaml
+++ b/helm/csi-powerstore/values.yaml
@@ -87,6 +87,11 @@ controller:
   # Default value: None
   volumeNamePrefix: csivol
 
+  attacher:
+    image: k8s.gcr.io/sig-storage/csi-attacher:v4.2.0
+  provisioner:
+    image: k8s.gcr.io/sig-storage/csi-provisioner:v3.4.0
+
   # vgsnapshot: allows to configure volume-group-snapshot
   # volume-group-snapshot CRDs must be installed before installing driver
   vgsnapshot:
@@ -110,6 +115,7 @@ controller:
     #   false: disable volume snapshot feature(do not install snapshotter sidecar)
     # Default value: None
     enabled: true
+    image: k8s.gcr.io/sig-storage/csi-snapshotter:v6.2.1
 
     # snapNamePrefix: Prefix to apply to the names of a created snapshots
     # Allowed values: string
@@ -123,6 +129,7 @@ controller:
     #   false: disable volume expansion feature(do not install resizer sidecar)
     # Default value: true
     enabled: true
+    image: k8s.gcr.io/sig-storage/csi-resizer:v1.7.0
 
   healthMonitor:
     # enabled: Enable/Disable health monitor of CSI volumes
@@ -131,6 +138,7 @@ controller:
     #   false: disable checking of health condition of CSI volumes
     # Default value: false
     enabled: false
+    image: gcr.io/k8s-staging-sig-storage/csi-external-health-monitor-controller:v0.8.0
 
     # interval: Interval of monitoring volume health condition
     # Allowed values: Number followed by unit (s,m,h)
@@ -164,7 +172,8 @@ controller:
     replicationPrefix: "replication.storage.dell.com"
 
   # Image for csi-metadata-retriever
-  metadataretriever: dellemc/csi-metadata-retriever:v1.3.0
+  metadataretriever:
+    image: dellemc/csi-metadata-retriever:v1.3.0
 
   # nodeSelector: Define node selection constraints for controller pods.
   # For the pod to be eligible to run on a node, the node must have each
@@ -204,6 +213,9 @@ node:
   # Default value: None
   nodeIDPath: /etc/machine-id
 
+  registrar:
+    image: k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.6.3
+
   healthMonitor:
     # enabled: Enable/Disable health monitor of CSI volumes- volume usage, volume condition
     # Allowed values:
@@ -278,6 +290,12 @@ connection:
   # Default value: false
   enableCHAP: false
 
+storageClasses:
+
+secrets:
+  create: true
+  arrays:
+
 # CSI driver log level
 # Allowed values: "error", "warn"/"warning", "info", "debug", "error"
 # Default value: "debug"
@@ -336,6 +354,5 @@ podmon:
   #    - "--driverPodLabelValue=dell-storage"
   #    - "--ignoreVolumelessPods=false"
 
-images:
-  # "driver" defines the container image, used for the driver container.
-  driverRepository: dellemc
+driver:
+  image: dellemc/csi-powerstore:v2.6.0
-- 
2.25.1
